README.md

```diff
@@ -40,8 +40,8 @@ For Ginkgo core library:
 *   C++14 compliant compiler, one of:
     *   _gcc 5.3+, 6.3+, 7.3+, all versions after 8.1+_
     *   _clang 3.9+_
-    *   _Intel compiler 2017+_
-    *   _Apple LLVM 8.0+_ (__TODO__: verify)
```

Comment:
Only partially relevant to this commit, but when updating versions one could mention clang-format < 11.

Reply:
Good point, we only mention a lower version for clang-format. Maybe we can manage to fix that with a slightly modified format, though.